haystack/nodes/prompt/invocation_layer/amazon_bedrock.py

```diff
@@ -0,0 +1,368 @@
+
+        model_available = model_name_or_path in available_model_ids
+        if not model_available:
+            raise AmazonBedrockConfigurationError(
+                f"The model {model_name_or_path} is not available in Amazon Bedrock. "
+                f"Please make sure the model is available in the configured AWS region and you've been granted access."
+            )
+
+        stream: bool = kwargs.get("stream", False)
+        model_supports_streaming = model_name_or_path in model_ids_supporting_streaming
+        if stream and not model_supports_streaming:
+            raise AmazonBedrockConfigurationError(
+                f"The model {model_name_or_path} does not offer streaming support. "
+                f"Please remove the `stream` parameter."
+            )
+
+        return model_supported
+
+    def invoke(self, *args, **kwargs):
+        kwargs = kwargs.copy()
+        prompt: str = kwargs.pop("prompt", None)
+        stream: bool = kwargs.get("stream", self.model_adapter.model_kwargs.get("stream", False))
+
+        if not prompt or not isinstance(prompt, (str, list)):
+            raise ValueError(
+                f"No valid prompt provided. Model {self.model_name_or_path} requires a valid prompt."
+                f"Make sure to provide a prompt in the format that the model expects."
+            )
+
+        body = self.model_adapter.prepare_body(prompt=prompt, **kwargs)
+        try:
+            if stream:
+                response = self.client.invoke_model_with_response_stream(
+                    body=json.dumps(body),
+                    modelId=self.model_name_or_path,
+                    accept="application/json",
+                    contentType="application/json",
+                )
+                response_stream = response["body"]
+                handler: TokenStreamingHandler = kwargs.get(
+                    "stream_handler",
+                    self.model_adapter.model_kwargs.get("stream_handler", DefaultTokenStreamingHandler()),
+                )
+                responses = self.model_adapter.get_stream_responses(stream=response_stream, stream_handler=handler)
+            else:
+                response = self.client.invoke_model(
+                    body=json.dumps(body),
+                    modelId=self.model_name_or_path,
+                    accept="application/json",
+                    contentType="application/json",
+                )
+                response_body = json.loads(response.get("body").read().decode("utf-8"))
+                responses = self.model_adapter.get_responses(response_body=response_body)
+        except ClientError as exception:
+            raise AmazonBedrockInferenceError(
+                f"Could not connect to Amazon Bedrock model {self.model_name_or_path}. "
+                "Make sure your AWS environment is configured correctly, "
+                "the model is available in the configured AWS region and you've been granted access."
```

Comment:
```suggestion
                "the model is available in the configured AWS region, and you have access."
```
again - access to the model, region? how do they check the config? any link to docs?